### PR TITLE
[Preprocessing] Crop preprocessing support

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
 - Crop preprocessing step support
 - Note: instead of 'ov::Coordinate' (as originally proposed in ticket) simple std::vector<int> is used. This is because Coordinate don't support negative dimensions
 - Python bindings

### Tests:
 - Added unit tests, template reference tests, cpu and gpu tests

### Tickets:
 - 71453

### CODE REVIEW

File: docs/template_plugin/tests/functional/subgraph_reference/preprocess.cpp [hunk 1]
```diff
@@ -880,6 +880,45 @@ static RefPreprocessParams set_shape_with_resize() {
...
+    RefPreprocessParams res("preprocess_crop_basic");
+    res.function = []() {
+        auto f = create_simple_function(element::f32, PartialShape{2, 2, 2, 2});
+        auto p = PrePostProcessor(f);
+        p.input().tensor().set_shape({4, 4, 4, 4});
+        p.input().preprocess().crop({1, 1, 1, 1}, {-1, -1, -1, -1});
+        p.build();
+        return f;
+    };
+    auto input_size = 4 * 4 * 4 * 4;
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/subgraph_reference/preprocess.cpp:893]: Maybe:
ov::Shape shape{4, 4, 4, 4};
ov::shape_size(shape);

File: src/core/include/openvino/core/preprocess/preprocess_steps.hpp [hunk 2]
```diff
@@ -115,6 +115,20 @@ class OPENVINO_API PreProcessSteps final {
...
+    ///
+    /// \param begin Begin indexes for input tensor cropping. Negative values represent counting elements from the end
+    /// of input tensor
+    ///
+    /// \param end End indexes for input tensor cropping. End indexes are exclusive, which means values including end
+    /// edge are not included in the output slice. Negative values represent counting elements from the end of input
+    /// tensor
+    ///
+    /// \return Reference to 'this' to allow chaining with other calls in a builder-like manner.
+    PreProcessSteps& crop(const std::vector<int>& begin, const std::vector<int>& end);
```
Comment (ilyachur) [inline src/core/include/openvino/core/preprocess/preprocess_steps.hpp:130]: Can we use `ov::Coordinate`?

Comment (nosovmik) [reply src/core/include/openvino/core/preprocess/preprocess_steps.hpp:130]: `ov::Coordinate` is a vector<size_t> - it doesn't allow negative values, which is very convenient for cropping right/bottom sides

File: src/core/include/openvino/core/preprocess/preprocess_steps.hpp [hunk 3]
```diff
@@ -115,6 +115,20 @@ class OPENVINO_API PreProcessSteps final {
...
 
+    /// \brief Crop input tensor between begin and end coordinates. Under the hood, inserts `opset8::Slice` operation to
+    /// execution graph. It is recommended to use to together with `ov::preprocess::InputTensorInfo::set_shape` to set
+    /// original input shape before cropping
+    ///
+    /// \param begin Begin indexes for input tensor cropping. Negative values represent counting elements from the end
+    /// of input tensor
+    ///
+    /// \param end End indexes for input tensor cropping. End indexes are exclusive, which means values including end
+    /// edge are not included in the output slice. Negative values represent counting elements from the end of input
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/preprocess/preprocess_steps.hpp:126]: "End indexes are exclusive" is it aligned with ov::Tensor(t, roi) ?

File: src/tests/functional/plugin/gpu/shared_tests_instances/subgraph_tests/preprocess.cpp [hunk 4]
```diff
@@ -29,6 +29,8 @@ inline std::vector<preprocess_func> GPU_smoke_preprocess_functions() {
         preprocess_func(resize_linear_nhwc, "resize_linear_nhwc", 0.01f),
         preprocess_func(resize_cubic, "resize_cubic", 0.01f),
         preprocess_func(resize_dynamic, "resize_dynamic", 0.01f, { ov::Shape {1, 3, 123, 123} }),
+        preprocess_func(crop_basic, "crop_basic", 0.01f),
+        preprocess_func(crop_negative, "crop_negative", 0.01f),
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/gpu/shared_tests_instances/subgraph_tests/preprocess.cpp:33]: why threshold is not zero?

File: src/core/include/openvino/core/preprocess/preprocess_steps.hpp (hunk 3 above)
Comment (nosovmik) [reply src/core/include/openvino/core/preprocess/preprocess_steps.hpp:126]: Yes, if tensor is `{10,20,30,40,50}` then, ov::Tensor(tensor, Coordinate{1}, {Coordinate{4}) will give you {20,30,40}.
Also it is aligned with python data[begin:end] syntax + with 'Slice' operation

File: src/tests/functional/plugin/gpu/shared_tests_instances/subgraph_tests/preprocess.cpp (hunk 4 above)
Comment (nosovmik) [reply src/tests/functional/plugin/gpu/shared_tests_instances/subgraph_tests/preprocess.cpp:33]: just followed old style :-) will change if there is no more comments

**Comment by ilya-lavrenov** [comment]
@vladimir-paramuzov please, check whether this solution satisfies GPU usecase.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**